lib/modules/dosomething/dosomething_api/resources/campaign_resource.inc

```diff
@@ -352,6 +352,25 @@ function _campaign_resource_reportback($nid, $values) {
     watchdog('dosomething_api', '_campaign_resource_reportback:' . json_encode($values));
   }
 
-  return dosomething_reportback_save($values, $user);
+  $transaction_id = $_SERVER['HTTP_X_REQUEST_ID'];
+
+  if (! $transaction_id) {
```

Comment:
can you add some comments here to explain why we are only sending the reportback if there isn't a transaction id